src/main/java/ro/hiringsystem/mapper/JobApplicationMapper.java

```diff
@@ -0,0 +1,21 @@
+package ro.hiringsystem.mapper;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+import ro.hiringsystem.model.JobApplication;
+import ro.hiringsystem.model.dto.JobApplicationDto;
+
+@Mapper
+public interface JobApplicationMapper {
+
+    JobApplicationMapper INSTANCE = Mappers.getMapper(JobApplicationMapper.class);
+
+    @Mapping(source = "candidateUser", target = "candidateUser")
```

Comment:
These @Mapping annotation here are redundant, since the fields are implicitly mapped that way. So I think you could remove them.